proto/public/api.proto

```diff
@@ -124,8 +124,62 @@ message ApiError {
   string error = 1;
 }
 
+message ResourceSpec {
+  string namespace = 1;
+  string resource = 2;
+}
+
+message ResourceSelection {
+  ResourceSpec spec = 1;
+  string label_selector = 2;
+}
+
+message ResourceNotFound {
+  ResourceSpec spec = 1;
+  ApiError error = 2;
+}
+
+message StatSummaryRequest {
+  ResourceSelection resource = 1;
+  TimeWindow time_window = 2;
+
+  oneof outbound {
+    Empty none = 3;
+    ResourceSpec out_to_resource   = 4;
+    ResourceSpec out_from_resource = 5;
+  }
+}
+
+message StatSummaryResponse {
+  repeated StatTable stat_tables = 1;
+  ResourceNotFound error = 2;
+}
```

Comment:
Is it possible for the service to respond with multiple errors (i.e. if multiple specified resources do not exist)?

Reply:
The way I was thinking about it is that if we send requests for multiple resources, we'd  first parse a request like `conduit stat namespaces, deploys` into one `StatSummaryRequest` per resource.  So the response would only have one requested resource pertaining to it (though you can do thinks like, stats for deploys by pod, etc) but since we'd be getting things like pods per deployment from the k8s api, we wouldn't encounter non-existing resources at that point.

If we were thinking of moving the parsing of the resource string further back, and 
 the `ResourceSpec`'s `resource` could be a string like "namespaces,deployments", then I agree, the service could respond with multiple selected resources being missing

Reply:
@rmars that's fair!  in that case, should these fields be `oneof` i.e. so that it's an error for both `stat_tables` and `error` to be set?

Reply:
ooh, yeah, good idea.

Reply:
also, I think in this case, the `StatTable` doesn't need to be a repeated field

Reply:
What if the resource is `all`?

Reply:
ahhh, yeah, true. hmmm, you can't have repeated fields under a `oneof`. Would you rather we don't use the `oneof` here? or place it somewhere else?

Reply:
@rmars in that case i tend to do things like:

```protobuf
message StatSummaryResponse {
  oneof result {
    Ok ok = 1;
    ResourceNotFound error = 2;

  message Ok {
    repeated StatTable stat_tables
  }
}
```